### PR TITLE
[tools] Validate code excerpt configuration

### PR DESCRIPTION
**Pull request by stuartmorgan**
Ensures that if a REAME uses code excerpting, the package is configured
to drive it. This avoids the (so far common) mistake of adding tags when
the block check fails, but not setting up the package so that they
actually work.

Also adds a link to the wiki for the missing excerpt tag error so that
people find the instructions sooner, rather than adding tags via
guess-and-check.

Fixes https://github.com/flutter/flutter/issues/109231

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by tarrinneal — COMMENTED** [review]
As far as I can tell, this looks good. I'll let [bparrishMines](https://github.com/bparrishMines) give the approval, since I'm still not sure I know what all to check for.

File: script/tool/test/util.dart [hunk 1]
```diff
@@ -23,6 +23,13 @@ import 'mocks.dart';
 
 export 'package:flutter_plugin_tools/src/common/repository_package.dart';
 
+/// The relative path from a package to the file that is used to enable
+/// README excerpting for a package.
+// This is a shared constant to ensure that both readme-check and
+// update-excerpt are looking for the same file, so that readme-check can't
+// get out of sync with what actually drives excerpting.
+const String kReadmeExcerptConfigPath = 'example/build.excerpt.yaml';
```
Comment (bparrishMines) [inline script/tool/test/util.dart:31]: nit: I don't think we typically start constants with a `k` in Dart. I'm assuming this the pattern for this package?

**Review by bparrishMines — APPROVED** [review]
LGTM

File: script/tool/test/util.dart (hunk 1 above)
Comment (stuartmorgan) [reply script/tool/test/util.dart:31]: I learned recently that the Flutter style guide disagrees with Dart style on this for global constants:

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#begin-global-constant-names-with-prefix-k

**Pull request merged**